### ISSUE TITLE
FileHeader Rule throwing exception

### ISSUE THREAD

**Issue by zelein** [palantir/tslint#1694]
### Bug Report

- __TSLint version__: `3.15.*`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: `CLI`

#### Stack Trace
```
TypeError: Cannot read property 'search' of undefined
    at FileHeaderWalker.visitSourceFile (\node_modules\tslint\lib\rules\fileHeaderRule.js:61:28)
    at FileHeaderWalker.SyntaxWalker.visitNode (\node_modules\tslint\lib\language\walker\syntaxWalker.js:414:22)
    at FileHeaderWalker.SyntaxWalker.walk (\node_modules\tslint\lib\language\walker\syntaxWalker.js:7:14)
    at Rule.AbstractRule.applyWithWalker (\node_modules\tslint\lib\language\rule\abstractRule.js:19:16)
    at Rule.apply (\node_modules\tslint\lib\rules\fileHeaderRule.js:17:21)
    at Linter.lint (\node_modules\tslint\lib\tslint.js:68:37)
    at processFile (\node_modules\tslint\lib\tslint-cli.js:138:29)
    at \node_modules\tslint\lib\tslint-cli.js:177:74
    at Array.forEach (native)
    at Object.<anonymous> (\node_modules\tslint\lib\tslint-cli.js:177:41)
```

```ts
    FileHeaderWalker.prototype.visitSourceFile = function (node) {
        if (this.headerRegexp) {
            var text = node.getFullText();
            var offset = 0;
            if (text.indexOf("#!") === 0) {
                offset = text.indexOf("\n") + 1;
                text = text.substring(offset);
            }
            var match = text.match(this.commentRegexp);
            if (!match) {
                this.addFailure(this.createFailure(offset, 0, Rule.FAILURE_STRING));
            }
            else {
                var comment = match[2] ? match[2] : match[3];
                if (comment.search(this.headerRegexp) < 0) { // FAILING LINE
                    this.addFailure(this.createFailure(offset, 0, Rule.FAILURE_STRING));
                }
            }
        }
    };
```

#### Actual behavior

This is being executed on a file that does not currently contain a file header. The file has code in it, but the entirety of the file is commented out.

**Comment by adidahiya** [comment]
looks like a logic bug. accepting bugfix PRs!

**Comment by andy-hanson** [comment]
Looks like this was fixed in #1913.
